### PR TITLE
Helping define the PR description field

### PR DESCRIPTION
**Pull request by john-science**
## What is the change?

I am trying to make it more clear what the PR description should be, and point out that it is mandatory.

## Why is the change being made?

Based on recent PRs, I am hoping improving this template will increase the quality of our PRs.

---

## Checklist

- [X] This PR has only one purpose or idea.
- [X] Tests have been added/updated to verify that the new/changed code works.

- [X] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [X] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [X] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any important changes.
- [X] The documentation is still up-to-date in the `doc` folder.
- [X] The dependencies are still up-to-date in `setup.py`.

### CODE REVIEW

**Review by opotowsky — APPROVED** [review]
This looks good!

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 ## Description
 
-<!-- Mandatory: Include a summary of the change and link to any related GitHub Issues.-->
+<!-- MANDATORY: Include a summary of your change:
+1. What is your change?
+2. Why is your change being made?
+3. Link to any related GitHub Issues
+-->
```
[the remark this answers is not in the document]
Comment (john-science) [reply .github/pull_request_template.md:7]: That's an interesting idea.  Let me play with it.

[the remark this answers is not in the document]
Comment (john-science) [reply .github/pull_request_template.md]: Okay, try that one.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,6 +1,16 @@
...
+
+<!-- MANDATORY: Include a summary of your change -->
+
+## Why is the change being made?
+
+<!-- MANDATORY: And Link to any related GitHub Issues -->
+
+
+
+<!-- Optional: Link to any related GitHub Issues -->
```
Comment (opotowsky) [inline .github/pull_request_template.md:11]:
```suggestion

```

Comment (opotowsky) [reply .github/pull_request_template.md:11]: Not sure what you intended here, but suggesting some cleanup

Comment (opotowsky) [reply .github/pull_request_template.md:11]: oh!

in the `mandatory` above it also mentions Github  Issue links, and I was just deleting one. But this is fine, reject this and clean up the other one!

**Review by albeanth — APPROVED** [review]

**Review by opotowsky — APPROVED** [review]

**Pull request merged**